src/WWT.Providers/Providers/DSSProvider.cs

```diff
@@ -1,70 +1,65 @@
 using System;
 using System.Configuration;
-using System.IO;
 using WWTWebservices;
 
 namespace WWT.Providers
 {
     public class DSSProvider : RequestProvider
     {
-        public override void Run(WwtContext context)
+        private readonly IPlateTilePyramid _plateTile;
+
+        public DSSProvider(IPlateTilePyramid plateTile)
         {
-            string wwtTilesDir = ConfigurationManager.AppSettings["WWTTilesDir"];
-            string dssTerapixelDir = ConfigurationManager.AppSettings["DssTerapixelDir"];
+            _plateTile = plateTile;
+        }
 
+        public override void Run(WwtContext context)
+        {
             string query = context.Request.Params["Q"];
             string[] values = query.Split(',');
             int level = Convert.ToInt32(values[0]);
             int tileX = Convert.ToInt32(values[1]);
             int tileY = Convert.ToInt32(values[2]);
 
-            int octsetlevel = level;
-            string filename;
-
-
             if (level > 12)
             {
                 context.Response.Write("No image");
                 context.Response.Close();
-                return;
             }
-
-            if (level < 8)
+            else if (level < 8)
             {
+                string wwtTilesDir = ConfigurationManager.AppSettings["WWTTilesDir"];
+
                 context.Response.ContentType = "image/png";
-                Stream s = PlateTilePyramid.GetFileStream(wwtTilesDir + "\\dssterrapixel.plate", level, tileX, tileY);
-                int length = (int)s.Length;
-                byte[] data = new byte[length];
-                s.Read(data, 0, length);
-                context.Response.OutputStream.Write(data, 0, length);
-                context.Response.Flush();
-                context.Response.End();
-                return;
+
+                using (var s = _plateTile.GetStream(wwtTilesDir, "dssterrapixel.plate", level, tileX, tileY))
+                {
+                    s.CopyTo(context.Response.OutputStream);
+                    context.Response.Flush();
+                    context.Response.End();
+                }
             }
             else
             {
-                int L = level;
-                int X = tileX;
-                int Y = tileY;
-                string mime = "png";
-                int powLev5Diff = (int)Math.Pow(2, L - 5);
-                int X32 = X / powLev5Diff;
-                int Y32 = Y / powLev5Diff;
-                filename = string.Format(dssTerapixelDir + @"\DSS{0}L5to12_x{1}_y{2}.plate", mime, X32, Y32);
+                int powLev5Diff = (int)Math.Pow(2, level - 5);
```

Comment:
These changes are mostly to clean up duplication of variables that cause unnecessary confusion.